code/modules/mod/modules/modules_timeline.dm

```diff
@@ -0,0 +1,447 @@
+	density = FALSE
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	move_resist = INFINITY
+	interaction_flags_atom = NONE
+	///mob being eradicated by this field
+	var/mob/living/captured
+	///linked module. while this exists, the field will progress towards eradication. while it isn't, the field progresses away until it disappears. see attached for a special case
+	var/obj/item/mod/module/tem/tem
+	///time in seconds before someone is eradicated, assuming progress isn't interrupted
+	var/timetokill = 30
+	///the eradication appearance
+	var/mutable_appearance/mob_underlay
+	///the actual frame the animation is at in eradication, only changing when the progress towards eradication progresses enough to move to the next frame.
+	var/RPpos = null
+	///if a tem to link to isn't provided initially, this chrono field will progress towards eradication by itself without one.
+	var/attached = TRUE
+
+/obj/structure/chrono_field/Initialize(mapload, mob/living/target, obj/item/mod/module/tem/tem)
+	if(isliving(target))
+		if(!tem)
+			attached = FALSE
+		target.forceMove(src)
+		captured = target
+		var/icon/mob_snapshot = getFlatIcon(target)
+		var/icon/cached_icon = new()
+
+		for(var/i in 1 to CHRONO_FRAME_COUNT)
+			var/icon/removing_frame = icon('icons/obj/chronos.dmi', "erasing", SOUTH, i)
+			var/icon/mob_icon = icon(mob_snapshot)
+			mob_icon.Blend(removing_frame, ICON_MULTIPLY)
+			cached_icon.Insert(mob_icon, "frame[i]")
+
+		mob_underlay = mutable_appearance(cached_icon, "frame1")
+		update_appearance()
+
+		desc = initial(desc) + "<br>[span_info("It appears to contain [target.name].")]"
+	START_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/structure/chrono_field/Destroy()
+	if(tem)
+		tem.field_disconnect(src)
+	return ..()
+
+/obj/structure/chrono_field/update_overlays()
+	. = ..()
+	var/ttk_frame = 1 - (timetokill / initial(timetokill))
+	ttk_frame = clamp(CEILING(ttk_frame * CHRONO_FRAME_COUNT, 1), 1, CHRONO_FRAME_COUNT)
+	if(ttk_frame != RPpos)
+		RPpos = ttk_frame
+		underlays -= mob_underlay
+		mob_underlay.icon_state = "frame[RPpos]"
+		underlays += mob_underlay
+
+/obj/structure/chrono_field/process(delta_time)
+	if(!captured)
+		qdel(src)
+		return
+
+	if(timetokill > initial(timetokill))
+		for(var/atom/movable/freed_movable in contents)
+			freed_movable.forceMove(drop_location())
+		qdel(src)
+	else if(timetokill <= 0)
+		to_chat(captured, span_boldnotice("As the last essence of your being is erased from time, you are taken back to your most enjoyable memory. You feel happy..."))
+		var/mob/dead/observer/ghost = captured.ghostize(1)
+		if(captured.mind)
+			if(ghost)
+				ghost.mind = null
+		qdel(captured)
+		qdel(src)
+	else
+		captured.Unconscious(80)
+		if(captured.loc != src)
+			captured.forceMove(src)
+		update_appearance()
+		if(tem)
+			if(tem.field_check(src))
+				timetokill -= delta_time
+			else
+				tem = null
+				return .()
```

Comment:
Son I am fucking disappoint

Reply:
`return` would be identical to `return .()`, it's `process()`

Reply:
hey don't look at me it's oldcode

Reply:
i mean i'll fix it though